ui/articles-aem-renderer/src/main/java/org/cru/godtools/articles/aem/service/support/ArticleParser.java

```diff
@@ -62,16 +65,20 @@
             if (isArticleOrCategory(nextKey)) {
                 JSONObject returnedObject = evaluatingObject.optJSONObject(nextKey);
                 if (returnedObject != null) {
+                    urlParams.add(nextKey);
```

Comment:
if you don't remove the `nextKey` urlParams you add here though you will have a corrupted url.

Reply:
using a `Uri` object you can just use this line of code: `final Uri newUrl = url.buildUpon().appendPath(nextKey).build();`